katsdpimager/primary_beam.py

```diff
@@ -0,0 +1,252 @@
+"""Primary beam models and correction."""
+
+from abc import ABC, abstractmethod
+import math
+from typing import Optional, Sequence, Mapping, Any
+
+import numpy as np
+import h5py
+import pkg_resources
+import astropy.units as units
+
+
+class BeamRangeError(ValueError):
+    """Requested data lies outside the beam model."""
```

Comment:
I'm thinking that a better solution (or at least an alternative) should be to return NaNs outside the coverage of the model. That allows the full extent of a circular model to be retrieved, and for the current use case, all I really care about is that stuff outside the main lobe gets blanked.

Reply:
Yes, I agree. Also then blanking frequencies that may lie outside of the band associated with the PB as well?

Reply:
Yes, sounds consistent.